umh-core/pkg/generated/status-message.go

```diff
@@ -0,0 +1,317 @@
+// Copyright 2025 UMH Systems GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This file was generated from JSON Schema using quicktype, do not modify it directly.
+// To parse and unparse this JSON data, add this code to your project and do:
+//
+//    statusMessage, err := UnmarshalStatusMessage(bytes)
+//    bytes, err = statusMessage.Marshal()
+
+package generated
+
+import "bytes"
+import "errors"
+
+import "encoding/json"
+
+func UnmarshalStatusMessage(data []byte) (StatusMessage, error) {
+	var r StatusMessage
+	err := json.Unmarshal(data, &r)
+	return r, err
+}
+
+func (r *StatusMessage) Marshal() ([]byte, error) {
+	return json.Marshal(r)
+}
+
+// Schema for the status message containing system state and DFC information
+type StatusMessage struct {
+	Core    Core                   `json:"core"`
+	General General                `json:"general"`
+	Plugins map[string]interface{} `json:"plugins"`
+}
+
+type Core struct {
+	Agent             Agent                  `json:"agent"`
+	Container         Container              `json:"container"`
+	Dfcs              []Dfc                  `json:"dfcs"`
+	EventsTable       map[string]EventsTable `json:"eventsTable,omitempty"`
+	Latency           Latency                `json:"latency"`
+	Redpanda          Redpanda               `json:"redpanda"`
+	ReleaseChannel    string                 `json:"releaseChannel"`
+	SupportedFeatures []string               `json:"supportedFeatures"`
+	UnsTable          map[string]UnsTable    `json:"unsTable,omitempty"`
+	Version           string                 `json:"version"`
+}
+
+type Agent struct {
+	Health Health `json:"health"`
+}
+
+type Health struct {
+	Message string `json:"message"`
+	State   string `json:"state"`
+}
+
+type Container struct {
+	Architecture string `json:"architecture"`
+	CPU          CPU    `json:"cpu"`
+	Disk         Disk   `json:"disk"`
+	Hwid         string `json:"hwid"`
+	Memory       Memory `json:"memory"`
+}
+
+type CPU struct {
+	Health Health  `json:"health"`
+	Limit  float64 `json:"limit"`
+	Usage  float64 `json:"usage"`
+}
+
+type Disk struct {
+	Health Health  `json:"health"`
+	Limit  float64 `json:"limit"`
+	Usage  float64 `json:"usage"`
+}
+
+type Memory struct {
+	Health Health  `json:"health"`
+	Limit  float64 `json:"limit"`
+	Usage  float64 `json:"usage"`
+}
+
+type Dfc struct {
+	Connections        []Connection `json:"connections,omitempty"`
```

Comment:
maybe it would make sense to directly include something like "desiredState"? so that you can directly see when an instance is "paused", and maybe it is then quite easy to implement a pause action